doc/configuration.md

```diff
@@ -0,0 +1,312 @@
+2. Merges `.d` files in alphabetical order:
+   - `01-network.toml`
+   - `02-ssh.toml`
+   - `99-overrides.toml`
+3. Applies any CLI overrides if present.
+4. Fills in missing values with defaults from `config.rs`.
+
+### Example 3: Default Path without --config
+**Assumption:** If no `--config` path is specified, `azure-init` will only use the default values set within `config.rs`.
+
+**Order of Loading:**
+
+1. Loads configuration directly from defaults specified in `config.rs`.
+
+## Validation and Deserialization Process
+
+Azure Init uses strict validation on configuration fields to ensure they match expected types and values. For fields defined as enums or specific types, the system checks that values provided by the user align with supported options. If a configuration includes an unsupported value or incorrect type, deserialization will fail.
+
+### Configuration Struct and Field Definitions
+
+- The `Config` struct in Azure Init includes fields that are defined with specific types or enums, like `bool` for `query_sshd_config`.
+- Each field is expected to be of a specific type when deserialized by Serde (the library used for parsing configuration files).
+
+### Error Handling During Deserialization
+- When `Config::load()` attempts to load a configuration file (either from a specified file, directory, or default
+path), it reads the file content and uses `toml::from_str` to convert it into a `Config` struct.
+
+- If a field value in the configuration file doesn’t match its expected type (e.g., if `query_sshd_config` is set to `"not_a_boolean"` instead of `true` or `false`), Serde encounters a deserialization error because it can’t map the provided value to the expected type.
+
+### Propagation of Deserialization Errors 
+ - When deserialization fails, an error is generated and logged by `Config::load_from_file`, indicating that the configuration file could not be parsed correctly. This error is wrapped in an `Error::Io` type and returned by the function.
+- This `Error::Io` is propagated up the call stack to the `Config::load()` function, which in turn returns the error to the caller.
+- In the main application, `Config::load()` is called with `.expect("Failed to load configuration")`. As a result, any failure in loading the configuration causes the application to terminate immediately with a clear error message. This prevents the provisioning process from proceeding with an invalid configuration and provides immediate feedback to the user about the issue.
+
+
+
+### Example of an Unsupported Value
+
+Here’s an example configuration with an invalid value for `query_sshd_config`. This field expects a boolean (`true` or `false`), but in this case, an unsupported string value `"not_a_boolean"` is provided.
+
+```toml
+# Invalid value for query_sshd_config (not a boolean)
+[ssh]
+query_sshd_config = "not_a_boolean" # This will cause a validation failure
+```
+
+
+## Configuration Fields
+
```

Comment:
Thanks for your hard work.

Just curious, why do we have here in markdown documents such a huge list of configuration fields with every detail, including default value, etc?
Those are helpful, but in the long term not so maintainable. Think about, for example, a case of changing a default value from `true` to `false`. Then you would need to touch both code and the separate doc. 
I would prefer having such details in individual rustdocs of each struct, enum, etc. in actual `.rs` files.

Reply:
Originally, it was included as part of an ask by another dev to outline how Confg works and how  I chose the architecture I did for the config. There should be rustdocs for each of the structs that detail the default and the use for each struct in config.rs, secondary to this. But if you think this is confusing to have / maintain as well, I can always remove it!

Reply:
Yes, I understand that listing configs in the markdown docs could be meaningful during design phase or as draft pull request. However, keeping such details in separate documents does not look like an idiomatic Rust style. We should either move them to `.rs` files or simply remove.

Reply:
I removed those details from the document, along with many others that may be difficult to maintain.